src/main/java/trackr/model/task/TaskStatus.java

```diff
@@ -0,0 +1,69 @@
+package trackr.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static trackr.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Task's status in the task list.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTaskStatus(String)}
+ */
+public class TaskStatus {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Task status should only be `N` for not done or `D` for done";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
```

Comment:
Change to proper regex to check that only N or D is given
```suggestion
    public static final String VALIDATION_REGEX = "^[N|D]$";
```